### PR TITLE
Bump to scalafix 0.9.0-RC1

### PR DESCRIPTION
**Pull request by MasseGuillaume**
@olafurpg @julienrf 
tested against akka: https://github.com/MasseGuillaume/akka/commit/d34a9c13c58135a9d1133b91c9f29fae4acb71a3

### CODE REVIEW

**Comment by MasseGuillaume** [comment]
@olafurpg the instruction are almost identical, that's why I did not created two sections.

File: README.md [hunk 1]
```diff
@@ -62,11 +64,10 @@ libraryDependencies += collectionCompat // required for Collection213CrossCompat
 addCompilerPlugin(scalafixSemanticdb)
 scalacOptions ++= List(
   "-Yrangepos",
-  "-Ywarn-unused-import",
-  "-P:semanticdb:synthetics:on"
+  "-P:semanticdb:synthetics:on" // Required by the collection rewrites
 )
 
-scalafixDependencies in ThisBuild += "org.scala-lang.modules" %% "scala-collection-migrations" % Dependencies.collectionCompatVersion
+scalafixDependencies in ThisBuild += "org.scala-lang.modules" %% "scala-collection-migrations" % collectionCompatVersion
```
Comment (olafurpg) [inline README.md]: nitpick, is the variable for the version necessary? I think it's more copy-paste friendly if the module id is self-contained

**Comment by olafurpg** [comment]
@MasseGuillaume if you have separate sections then you can afford to explain in more depth the differences between the rules. The differences are small but the behavior of the rules is significant enough that I think it could be clearer separated. It's worth stressing that `Upgrade` makes it impossible to compile against 2.12 and earlier.

For example:

----

### Collections213Upgrade

The `Collections213Upgrade` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for applications such as web servers that don't need to cross-compile against multiple Scala versions.

```scala
// build.sbt
scalafixDependencies += "org.scala-lang.modules" ...
scalacOptions ++= List("-Yrangepos", "-P:semanticdb:synthetics:on")

// sbt shell
> scalafix Collections213Upgrade
```
### Collections213CrossCompat

The `Collections213Upgrade` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for libraries that are cross-published for multiple Scala versions. 

To cross-build for 2.12 and 2.11, an additional module `???` is required to provide missing extensions methods (???). This enables you to write your library using the latest 2.13 collections API while still supporting users on an older Scala version.


```scala
// build.sbt
scalafixDependencies += "org.scala-lang.modules" ...
libraryDependencies +=  "org.scala-lang.modules" ...
scalacOptions ++= List("-Yrangepos", "-P:semanticdb:synthetics:on")

// sbt shell
> scalafix Collections213CrossCompat
```

File: README.md [hunk 2]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
 ```
 
+### Collection213Upgrade
+
+The `Collection213Upgrade` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for applications such as web servers that don't need to cross-compile against multiple Scala versions.
+
 ```scala
-// build.sbt or project/Build.scala
+// build.sbt
+scalafixDependencies += "org.scala-lang.modules" ...
```
Comment (julienrf) [inline README.md]: Why do you truncate the expression?

File: README.md [hunk 3]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
 ```scala
 // project/plugins.sbt
 
-addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.7.0-RC1")
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.0-RC1")
 ```
 
+### Collection213Upgrade
+
+The `Collection213Upgrade` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for applications such as web servers that don't need to cross-compile against multiple Scala versions.
```
Comment (julienrf) [inline README.md]: Can we remove “such as web servers”?

File: README.md [hunk 4]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
+
 ```scala
-// build.sbt or project/Build.scala
+// build.sbt
```
Comment (julienrf) [inline README.md]: We should explain what these line do before starting the code block:

> Add the `scala-collection-migrations` artifact to your scalafix dependencies and make sure that the following `scalacOptions` are enabled:
> ~~~ scala
> // build.sbt
> …
> ~~~
>

File: README.md [hunk 5]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
-addCompilerPlugin(scalafixSemanticdb)
-scalacOptions ++= List(
-  "-Yrangepos",
-  "-Ywarn-unused-import",
-  "-P:semanticdb:synthetics:on"
-)
 
-scalafixDependencies in ThisBuild += "org.scala-lang.modules" %% "scala-collection-migrations" % Dependencies.collectionCompatVersion
-```
+The `Collections213CrossCompat` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for libraries that are cross-published for multiple Scala versions.
```
Comment (julienrf) [inline README.md]: s/without/with?

File: README.md [hunk 6]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
+
+To cross-build for 2.12 and 2.11, an additional module `scala-collection-compat` is required to provide missing extensions methods. This enables you to write your library using the latest 2.13 collections API while still supporting users on an older Scala version.
```
Comment (julienrf) [inline README.md]: I would replace the first sentence with:

> To cross-build for 2.12 and 2.11, the rewrite rule introduces a dependency on the `scala-collection-compat` module, which provides the syntax of 2.13 on 2.12 and 2.11.

File: README.md [hunk 7]
```diff
@@ -38,45 +38,56 @@ The library also adds backported versions of new collection types, currently `sc
...
 ```scala
 // project/plugins.sbt
 
-addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.7.0-RC1")
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.0-RC1")
 ```
 
+### Collection213Upgrade
+
+The `Collection213Upgrade` rewrite upgrades to the 2.13 collections without the ability to compile the code-base with 2.12 or later. This rewrite is suitable for applications that don't need to cross-compile against multiple Scala versions.
```
Comment (julienrf) [inline README.md]: > without the ability to compile the code-base with 2.12 **or later**

The “or later” part is wrong.

Comment (julienrf) [reply README.md]: @MasseGuillaume Do you plan to address this comment?

**Comment by SethTisue** [comment]
expected to be superseded by #156

File: README.md (hunk 7 above)
Comment (olafurpg) [reply README.md]: Fixed in #156

**Comment by olafurpg** [comment]
Superseded by #156

**Pull request closed without merging**